crates/ethereum/node/src/node.rs

```diff
@@ -304,12 +304,11 @@ where
     Node: FullNodeTypes<Types = Types>,
 {
     type EVM = EthEvmConfig;
-    type Executor = BasicBlockExecutorProvider<EthEvmConfig>;
 
     async fn build_evm(
         self,
         ctx: &BuilderContext<Node>,
-    ) -> eyre::Result<(Self::EVM, Self::Executor)> {
+    ) -> eyre::Result<(Self::EVM, BasicBlockExecutorProvider<Self::EVM>)> {
```

Comment:
do we even need to bake `BasicBlockExecutorProvider` into the API like that, if we can always construct this from Self::Evm

so we could even move `BasicBlockExecutorProvider::new` inside the trait itself?

because this now just acts as a proxy for `BasicBlockExecutorProvider::executor(`

so it feels like we can move the `fn executor` directly inside the `ConfigureEvm` trait perhaps?

Reply:
makes total sense, done here https://github.com/paradigmxyz/reth/pull/15989/commits/7f19447fba4920ae016c3777286024a4a4253305, instead of modifying `ConfigureEvm` I've used `BasicBlockExecutorProvider::new` while building components, the only place where it is needed, ptal